firecrest/AsyncClient.py

```diff
@@ -777,10 +850,76 @@ async def compress(self, machine: str, source_path: str, target_path: str) -> st
             additional_headers={"X-Machine-Name": machine},
             data={"targetPath": target_path, "sourcePath": source_path},
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
+            resp.status_code != 400 or
+            resp.json()["error"] != timeout_str
+        ):
+            self._json_response([resp], 201)
+        else:
+            logger.debug(
+                f"Compression of {source_path} to {target_path} has finished "
+                f"with timeout signal. Will submit a job to compress the "
+                f"file."
+            )
+            job_info = await self.submit_compress_job(
+                machine,
+                source_path,
+                target_path
+            )
+            jobid = job_info['jobid']
+            active_jobs = await self.poll_active(
+                machine,
+                [jobid]
+            )
+            intervals = itertools.cycle(
+                [5,  # We know it took at least 5 seconds
+                 1, 1, 1, 1, 1,
+                 5, 5, 5, 5,
+                 10, 10, 10,
+                 30]
+            )
+            while (
+                active_jobs and
+                not slurm_state_completed(active_jobs[0]['state'])
+            ):
+                await asyncio.sleep(next(intervals))
```

Comment:
We need to discuss this, I think it's better to not `sleep` here. But it requires more design change that we can discuss on Friday.

Reply:
Hm I think this point we didn't actually discuss. Is it still okay to leave this as it is?

Reply:
Makes sense to me to have a sleep here and not overload the F7T api